### PR TITLE
Finish npm and yarn impelementation with history align

### PR DESCRIPTION
**Pull request by khalifapro**

### CODE REVIEW

File: internal/helper/helper.go [hunk 1]
```diff
@@ -103,3 +106,68 @@ func GetCopyright(content string) string {
 
 	return ""
 }
+
+// GetCopyrightText ...
+func GetCopyrightText(path string) string {
```
Comment (niravpatel27) [inline internal/helper/helper.go:111]: Please use `GetCopyright` method.

File: internal/helper/helper.go [hunk 2]
```diff
@@ -103,3 +106,68 @@ func GetCopyright(content string) string {
...
+	ind := strings.Index(c, "Copyright (c)")
+	if ind < 0 {
+		return ""
+	}
+	copyWrite := strings.Split(c[ind:], "\n")
+	return strings.TrimSuffix(copyWrite[0], "\r")
+}
+
+// GetJSLicense ...
+func GetJSLicense(path string, pkName string, licenses map[string]string, modPath string, modManifest string) string {
```
Comment (niravpatel27) [inline internal/helper/helper.go:123]: Use GetLicenses method instead

**Comment by niravpatel27** [comment]
What happens if you dont find any dependencies for npm?  also, we need to make sure our error messages are consistent with other modules. plz review go module

File: internal/helper/helper.go (hunk 2 above)
Comment (khalifapro) [reply internal/helper/helper.go:123]: > What happens if you dont find any dependencies for npm? also, we need to make sure our error messages are consistent with other modules. plz review go module

confirmed! the two errors are consistent

**Comment by khalifapro** [comment]
confirmed! the two errors are consistent

File: internal/helper/helper.go [hunk 3]
```diff
@@ -99,6 +99,12 @@ func GetCopyright(content string) string {
 		if strings.Contains(strings.ToLower(tokens[0]), "copyright") {
 			return line
 		}
+		for  _,l :=  range lines {
```
Comment (niravpatel27) [inline internal/helper/helper.go:102]: Is it needed? if yes, then can we please combine. The func is doing the redundant  checks

File: internal/modules/npm/errors.go [hunk 4]
```diff
@@ -8,4 +8,7 @@ import (
 
 type errType error
 
-var errDependenciesNotFound errType = errors.New("Unable to generate SPDX file, no modules or vendors found. Please install them before running spdx-sbom-generator, e.g.: `npm install`")
+var (
+	errDependenciesNotFound errType = errors.New("unable to generate SPDX file, no modules or vendors found. Please install them before running spdx-sbom-generator, e.g.: `npm install`")
```
Comment (niravpatel27) [inline internal/modules/npm/errors.go:12]: Make error specific to npm. You dont need vendor word.

File: internal/modules/npm/handler.go [hunk 5]
```diff
@@ -66,17 +89,133 @@ func (m *npm) SetRootModule(path string) error {
...
+		r := ""
+		if d["resolved"] != nil {
+			r = d["resolved"].(string)
+			if strings.Contains(r, npmRegistry) {
+				mod.Supplier.Name = "NOASSERTION"
+			}
+		}
+
+		mod.PackageURL = r
+		if d["integrity"] != nil {
```
Comment (niravpatel27) [inline internal/modules/npm/handler.go:187]: Maybe move if-else to separate lookup func?

**Comment by niravpatel27** [comment]
Remove `todo` comments

**Review by niravpatel27 — APPROVED** [review]

**Pull request merged**